data-prepper-plugins/s3-source/src/main/java/com/amazon/dataprepper/plugins/source/configuration/OnErrorOption.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.source.configuration;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public enum OnErrorOption {
+    DELETE_MESSAGES("delete_messages"),
+    DLQ("dlq");
```

Comment:
I wonder if "retain_messages" would be clearer. That is really all it does. It is up to the user to configure SQS to use a DLQ.